modules/fedimint-ln-client/src/lib.rs

```diff
@@ -433,6 +433,85 @@ impl ClientModule for LightningClientModule {
     ) -> anyhow::Result<serde_json::Value> {
         cli::handle_cli_command(self, args).await
     }
+
+    async fn handle_rpc(
+        &self,
+        request: serde_json::Value,
+    ) -> BoxStream<'_, anyhow::Result<serde_json::Value>> {
+        Box::pin(try_stream! {
+            let request = serde_json::from_value(request)?;
+            match request {
+                LnClientRpcRequest::CreateBolt11Invoice {
+                    amount,
+                    description,
+                    expiry_time,
+                    extra_meta,
+                    gateway,
+                } => {
+                    let (op, invoice, _) = self
+                        .create_bolt11_invoice(
+                            amount,
+                            lightning_invoice::Bolt11InvoiceDescription::Direct(
+                                &lightning_invoice::Description::new(description)?,
+                            ),
+                            expiry_time,
+                            extra_meta,
+                            gateway,
+                        )
+                        .await?;
+                    yield serde_json::json!({
+                        "operation_id": op,
+                        "invoice": invoice,
+                    });
+                }
+                LnClientRpcRequest::PayBolt11Invoice {
+                    maybe_gateway,
+                    invoice,
+                    extra_meta,
+                } => {
+                    let outgoing_payment = self.pay_bolt11_invoice(
+                        maybe_gateway,
+                        invoice,
+                        extra_meta,
+                    ).await?;
+                    yield serde_json::to_value(outgoing_payment)?;
+                },
+                LnClientRpcRequest::SubscribeLnPay { operation_id } => {
+                    for await state in self.subscribe_ln_pay(operation_id).await?.into_stream() {
+                        yield serde_json::to_value(state)?;
```

Comment:
I see. Meh.

I regret any subscriptions in Fedimint code, including jsonrpc. E.g. this type of functionality can be handle better with a `GetNextLNPayEvent { cursor: Option<SomeEventId>, timeout: Option<Duration> }` kind of RPC.

What if someone wants to use these rpcs over http. Like what @Kodylow already has in `fedimint-http` or something. we should have `fedimint-http`-like thing, that is just a wrapper over these new rpcs. But now if we make things a stream here, what do we do? We either have to invent cursor-based schims to make it possible, or force jsonrpc and complexity on all downstream.

The opposite is actually great. If the inner-logic uses cursor-based API, then wrapping it around to get a stream is trivial.

Similar with all the client event logs operations, etc. Had they had cursor-based APIs, they would be easier to follow accross restarts, from external system. And also the complexity of handling them wouldn't suck you out of everything that touches them. :D